tests/test_frame.py

```diff
@@ -1955,6 +1955,27 @@ def test_types_rename() -> None:
     df.rename(columns=lambda s: s.upper())
 
 
+def test_types_rename_axis() -> None:
+    df = pd.DataFrame({"col_name": [1, 2, 3]})
+    df.index.name = "a"
+    df.columns.name = "b"
+
+    # Rename axes with `mapper` and `axis`
+    df.rename_axis("A")
```

Comment:
For those tests, can you use the `check(assert_type(...))` framework so that tests the resulting types on top of testing at run time that it works.
For that one it should look like `check(assert_type(df.rename_axis("A"), pd.DataFrame), pd.DataFrame)`.

Reply:
OK, I've done so in af01cc3.

Reply:
And simplified a bit in 7625017.

Reply:
I believe you may need to keep them line by line for the checks instead of a loop since they act both at the static level and runtime but @Dr-Irv could confirm that.

Reply:
OK, I'll hold off on changing until @Dr-Irv weighs in.

Reply:
@loicdiridollou is correct and I commented about that.

Reply:
OK: 7ced65d